pytest_html/plugin.py

```diff
@@ -127,18 +131,36 @@ def __lt__(self, other):
                      'XPassed', 'Skipped', 'Passed')
             return order.index(self.outcome) < order.index(other.outcome)
 
-        def append_extra_html(self, extra, additional_html, links_html):
+        def append_extra_html(self, extra):
             href = None
             if extra.get('format') == extras.FORMAT_IMAGE:
-                href = '#'
-                image = 'data:image/png;base64,{0}'.format(
-                        extra.get('content'))
-                additional_html.append(html.div(
-                    html.a(html.img(src=image), href="#"),
+                image_src = None
+                if self.contained_html:
+                    image_src = 'data:image/png;base64,{0}'.format(
+                            extra.get('content'))
+                    href = '#'
+                else:
+                    hash_key = self.test_id + self.outcome
```

Comment:
We want the hash to be unique, so the key should also be unique. I'm not sure what benefit adding the `outcome` to the key provides? The report may include multiple images, and we'd also want each image to have a unique key. I would suggest a combination of `test_id` and the index of the extra that we're currently handling. We may also need to consider reruns, so perhaps if it's available the index of the current rerun.